framework/framework.go

```diff
@@ -142,3 +142,60 @@ func (f *Framework) ProcessDelete(obj interface{}, resources []Resource) error {
 
 	return nil
 }
+
+// ProcessUpdate is a drop-in for an informer's UpdateFunc. It receives the
+// custom object observed during TPR watches and anything that implements
+// Resource. ProcessUpdate takes care about all necessary reconciliation logic
+// for update events. For complex resources this means state has to be created,
+// deleted and updated eventually, in this order.
+//
+//     func updateFunc(oldObj, newObj interface{}) {
+//         err := f.ProcessUpdate(oldObj, newObj, resources)
+//         if err != nil {
+//             // error handling here
+//         }
+//     }
+//
+//     newResourceEventHandler := &cache.ResourceEventHandlerFuncs{
+//         UpdateFunc:    updateFunc,
+//     }
+//
+func (f *Framework) ProcessUpdate(oldObj, newObj interface{}, resources []Resource) error {
```

Comment:
So what do you want now? Should the old custom object just be removed from `ProcessUpdate`?  I see your point and see how an operator can fail with the old custom object. I can see the operator fail as well though with the new custom object. Your example with an operator creating the deployment is a good example. How does the operator know how to find the deployment? The deployment has a name which is made up using information in the TPO. When the the information in the TPO change that are used to compute the deployment name then relying on the new TPO fails as well. So how to approach this? I we should use the old custom object to fetch the current state. In case this fails because a previous reconciliation failed the operator isn't functional anyway and it should retry until it succeeds. Then the operator either fails forever or eventually gets it right at some point. We probably have to go with some way and see how it goes. Feel free to chose which way to go now.